### PR TITLE
2.8.7

### PR DESCRIPTION
**Pull request by abr-ubiqube**

### CODE REVIEW

**Review by ArnaudFosse — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -261,16 +266,18 @@ services:
       msa-dev:
         condition: service_started
     environment:
-      <<: *es-credential
+      ES_MON: "yes"
```
Comment (UBiqube-ydu) [inline docker-compose.yml:269]: Not useful because the code is not merged yet. But we can keep it.

File: scripts/swarm-fix-route.sh [hunk 2]
```diff
@@ -152,14 +152,15 @@ function add_default_route {
...
-function add_514_nat_exception {
+function add_nat_exception {
   # $1 - namespace id
   # $2 - destinnation ip prefix
 
   sudo ip netns exec $1 iptables -t nat -nvL POSTROUTING --line-numbers > /dev/null
   if [ $? -eq 0 ]; then
     sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p udp --dport 514 -j ACCEPT
     sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p udp --dport 162 -j ACCEPT
+    sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p tcp --dport 6514 -j ACCEPT
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:163]: Missing TCP/514 :
sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p tcp --dport 514 -j ACCEPT

File: scripts/swarm-fix-route.sh [hunk 3]
```diff
@@ -225,6 +226,22 @@ function delete_514_nat_exception {
...
+      do
+        sudo ip netns exec $1 iptables -t nat -D POSTROUTING $rule
+        if [ $? -eq 0 ]; then
+          echo "NAT exception for dst tcp 6514 successfully deleted, rule num $rule"
+        else
+          echo "Error: Can't delete NAT exception. Can't continue."
+          exit 1
+        fi
+      done
+    fi
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:244]: Missing _tcp dpt:514_

File: scripts/swarm-fix-route.sh [hunk 4]
```diff
@@ -397,14 +414,14 @@ function main {
       a)
       echo "STEP 8:"
       echo "Updating <<$INGRESS_NS>> namespace..."
-      echo "---> adding NAT UDP 514 exception... "
-      add_514_nat_exception $INGRESS_NS $OVERLAY_NET_1_PREFIX
+      echo "---> adding NAT UDP/TCP UDP/514 162 TCP/6514 exception... "
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:417]: echo "---> adding NAT UDP:162 UDP:514 TCP:514 TCP:6514  exception..."

File: scripts/swarm-fix-route.sh [hunk 5]
```diff
@@ -397,14 +414,14 @@ function main {
...
+      add_nat_exception $INGRESS_NS $OVERLAY_NET_1_PREFIX
       echo ""
 
       echo "STEP 9:"
       echo "Updating load balancer <<$LB_NS_ID>> namespace..."
-      echo "---> adding NAT UDP 514 exception... "
-      add_514_nat_exception $LB_NS_ID $OVERLAY_NET_2_PREFIX
+      echo "---> adding NAT UDP/514 UDP/162 TCP/6514 exception... "
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:423]: echo "---> adding NAT UDP/514 UDP/162 TCP/514 TCP/6514 exception..."

File: scripts/swarm-fix-route.sh [hunk 6]
```diff
@@ -455,10 +472,10 @@ function main {
       s)
       echo "STEP 8:"
       echo "NAT EXCEPTIONS LIST:"
-      echo "---> show NAT UDP 514 exceptions in <<$INGRESS_NS>> namespace:"
+      echo "---> show NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$INGRESS_NS>> namespace:"
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:475]: echo "---> show NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$INGRESS_NS>> namespace:"

File: scripts/swarm-fix-route.sh [hunk 7]
```diff
@@ -455,10 +472,10 @@ function main {
...
       show_nat_exceptions $INGRESS_NS
       echo ""
-      echo "---> show NAT UDP 514 exceptions in <<$LB_NS_ID>> namespace:"
+      echo "---> show NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$LB_NS_ID>> namespace:"
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:478]: echo "---> show NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$LB_NS_ID>> namespace:"

File: scripts/swarm-fix-route.sh [hunk 8]
```diff
@@ -507,9 +524,9 @@ function main {
 
       d)
       echo "STEP 8:"
-      echo "Removing NAT UDP 514 exceptions in <<$INGRESS_NS>> namespace..."
+      echo "Removing NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$INGRESS_NS>> namespace..."
       delete_514_nat_exception $INGRESS_NS
-      echo "Removing NAT UDP 514 exceptions in <<$LB_NS_ID>> namespace..."
+      echo "Removing NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$LB_NS_ID>> namespace..."
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route.sh:529]: echo "Removing NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$LB_NS_ID>> namespace..."

File: scripts/swarm-fix-route_manager.sh [hunk 9]
```diff
@@ -154,14 +154,15 @@ function add_default_route {
...
-function add_514_nat_exception {
+function add_nat_exception {
   # $1 - namespace id
   # $2 - destinnation ip prefix
 
   sudo ip netns exec $1 iptables -t nat -nvL POSTROUTING --line-numbers > /dev/null
   if [ $? -eq 0 ]; then
     sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p udp --dport 514 -j ACCEPT
     sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p udp --dport 162 -j ACCEPT
+    sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p tcp --dport 6514 -j ACCEPT
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:165]: Missing TCP/514
sudo ip netns exec $1 iptables -t nat -I POSTROUTING 2 -m ipvs --ipvs -s 0.0.0.0/0 -d $2 -p tcp --dport 514 -j ACCEPT

File: scripts/swarm-fix-route_manager.sh [hunk 10]
```diff
@@ -227,6 +228,22 @@ function delete_514_nat_exception {
...
+        sudo ip netns exec $1 iptables -t nat -D POSTROUTING $rule
+        if [ $? -eq 0 ]; then
+          echo "NAT exception for dst tcp 6514 successfully deleted, rule num $rule"
+        else
+          echo "Error: Can't delete NAT exception. Can't continue."
+          exit 1
+        fi
+      done
+    fi
   fi
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:247]: Missing TCP/514

File: scripts/swarm-fix-route_manager.sh [hunk 11]
```diff
@@ -227,6 +228,22 @@ function delete_514_nat_exception {
...
+      do
+        sudo ip netns exec $1 iptables -t nat -D POSTROUTING $rule
+        if [ $? -eq 0 ]; then
+          echo "NAT exception for dst tcp 6514 successfully deleted, rule num $rule"
+        else
+          echo "Error: Can't delete NAT exception. Can't continue."
+          exit 1
+        fi
+      done
+    fi
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:246]: Missing TCP/514

File: scripts/swarm-fix-route_manager.sh [hunk 12]
```diff
@@ -399,18 +416,18 @@ function main {
       a)
       echo "STEP 8:"
       echo "Updating <<$INGRESS_NS>> namespace..."
-      echo "---> adding NAT UDP 514 exception... "
-      add_514_nat_exception $INGRESS_NS $OVERLAY_NET_1_PREFIX
+      echo "---> adding NAT UDP/514 UDP/162 TCP/6514 exception... "
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:419]: echo "---> adding NAT UDP/514 UDP/162 TCP/514 TCP/6514 exception... "

File: scripts/swarm-fix-route_manager.sh [hunk 13]
```diff
@@ -399,18 +416,18 @@ function main {
...
+      add_nat_exception $INGRESS_NS $OVERLAY_NET_1_PREFIX
       echo ""
       ;;
 
       s)
       echo "STEP 8:"
       echo "NAT EXCEPTIONS LIST:"
-      echo "---> show NAT UDP 514 exceptions in <<$INGRESS_NS>> namespace:"
+      echo "---> show NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$INGRESS_NS>> namespace:"
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:427]: echo "---> show NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$INGRESS_NS>> namespace:"

File: scripts/swarm-fix-route_manager.sh [hunk 14]
```diff
@@ -399,18 +416,18 @@ function main {
...
       show_nat_exceptions $INGRESS_NS
       echo ""
-      echo "---> show NAT UDP 514 exceptions in <<$LB_NS_ID>> namespace:"
+      echo "---> show NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$LB_NS_ID>> namespace:"
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:430]: echo "---> show NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$LB_NS_ID>> namespace:"

File: scripts/swarm-fix-route_manager.sh [hunk 15]
```diff
@@ -459,8 +476,8 @@ function main {
 
       d)
       echo "STEP 8:"
-      echo "Removing NAT UDP 514 exceptions in <<$INGRESS_NS>> namespace..."
-      delete_514_nat_exception $INGRESS_NS
+      echo "Removing NAT UDP/514 UDP/162 TCP/6514 exceptions in <<$INGRESS_NS>> namespace..."
```
Comment (UBiqube-ydu) [inline scripts/swarm-fix-route_manager.sh:479]: echo "Removing NAT UDP/514 UDP/162 TCP/514 TCP/6514 exceptions in <<$INGRESS_NS>> namespace..."

File: scripts/listServiceNodes.sh [hunk 16]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 # Get the swarm name if any
 DOCKER_STACK_NAME=''
-STACK_NAMES=$(docker stack ls --format {{.Name}} 2>/dev/null)
+STACK_NAMES=$(docker stack ls --orchestrator swarm --format {{.Name}} 2>/dev/null)
```
Comment (UBiqube-ydu) [inline scripts/listServiceNodes.sh:4]: _--orchestrator_ is a deprecated option, it should not be added.

**Review by amitraj83 — APPROVED** [review]

**Pull request merged**